periphery/src/Router.sol

```diff
@@ -105,29 +67,55 @@ contract Router {
         shares = lender.deposit(amount, msg.sender);
     }
 
-    function repayWithApprove(Lender lender, uint256 amount, address beneficiary) external returns (uint256 units) {
-        lender.asset().safeTransferFrom(msg.sender, address(lender), amount);
+    function repayWithPermit2(
+        Lender lender,
+        uint256 amount,
+        address beneficiary,
+        uint256 nonce,
+        uint256 deadline,
+        bytes calldata signature
+    ) external returns (uint256 units) {
+        // Transfer tokens from the caller to the lender.
+        PERMIT2.permitTransferFrom(
+            // The permit message.
+            IPermit2.PermitTransferFrom({
+                permitted: IPermit2.TokenPermissions({token: lender.asset(), amount: amount}),
+                nonce: nonce,
+                deadline: deadline
+            }),
+            // The transfer recipient and amount.
+            IPermit2.SignatureTransferDetails({to: address(lender), requestedAmount: amount}),
+            // The owner of the tokens, which must also be
+            // the signer of the message, otherwise this call
+            // will fail.
+            msg.sender,
+            // The packed signature that was the result of signing
+            // the EIP712 hash of `permit`.
+            signature
+        );
+
         units = lender.repay(amount, beneficiary);
     }
 
-    function repayWithPermit(
+    function isMaxRedeemDynamic(Lender lender, address owner) external view returns (bool) {
```

Comment:
When might this function be used?

Reply:
Used to determine whether a user's `maxRedeem` number of shares is changing. If it is, we may use the helper function on `Router` to redeem instead of plain `lender.redeem(...)`